### PR TITLE
Remove mTLS enablement via service annotation.

### PR DESCRIPTION
**Pull request by yangminzhu**
See #5826.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/conversion.go [hunk 1]
```diff
@@ -69,25 +65,12 @@ func convertLabels(obj meta_v1.ObjectMeta) model.Labels {
...
-	return meshconfig.AuthenticationPolicy_INHERIT
-}
-
 func convertPort(port v1.ServicePort, obj meta_v1.ObjectMeta) *model.Port {
 	return &model.Port{
 		Name:                 port.Name,
 		Port:                 int(port.Port),
 		Protocol:             ConvertProtocol(port.Name, port.Protocol),
-		AuthenticationPolicy: extractAuthenticationPolicy(port, obj),
+		AuthenticationPolicy: meshconfig.AuthenticationPolicy_INHERIT,
```
Comment (diemtvu) [inline pilot/pkg/serviceregistry/kube/conversion.go]: We can actually remove this AuthenticationPolicy from Port struct (model/service.go).

File: pilot/pkg/proxy/envoy/v1/mock/discovery.go [hunk 2]
```diff
@@ -17,8 +17,7 @@ package mock
 import (
 	"fmt"
 	"net"
-
-	meshconfig "istio.io/api/mesh/v1alpha1"
+	//meshconfig "istio.io/api/mesh/v1alpha1"
```
Comment (diemtvu) [inline pilot/pkg/proxy/envoy/v1/mock/discovery.go]: Remove

File: pilot/pkg/model/authentication.go [hunk 3]
```diff
@@ -42,19 +42,14 @@ func GetConsolidateAuthenticationPolicy(mesh *meshconfig.MeshConfig, store Istio
 		}
 	}
 
-	legacyPolicy := consolidateAuthPolicy(mesh, port.AuthenticationPolicy)
+	legacyPolicy := meshAuthPolicy(mesh)
```
Comment (diemtvu) [inline pilot/pkg/model/authentication.go]: You can remove meshAuthPolicy function, and use `mesh.AuthPolicy` directly.

File: pilot/pkg/proxy/envoy/v1/mock/discovery.go (hunk 2 above)
Comment (yangminzhu) [reply pilot/pkg/proxy/envoy/v1/mock/discovery.go]: Thanks for catching this!

File: pilot/pkg/model/authentication.go (hunk 3 above)
Comment (yangminzhu) [reply pilot/pkg/model/authentication.go]: Refactored to directly use legacyAuthenticationPolicyToPolicy(), let me know if you're thinking something else.

**Pull request merged**